torch/csrc/distributed/c10d/CUDASymmetricMemory.cu

```diff
@@ -592,12 +587,27 @@ int CUDASymmetricMemory::get_world_size() {
   return world_size_;
 }
 
+Block::Block(
+  c10::intrusive_ptr<AllocationRef> alloc_ref,
+  int device_idx,
+  size_t block_size,
+  size_t buffer_size,
+  size_t signal_pad_offset,
+  const std::optional<std::string>& group_name)
+    : alloc_ref(std::move(alloc_ref)),
+      device_idx(device_idx),
+      block_size(block_size),
+      buffer_size(buffer_size),
+      signal_pad_offset(signal_pad_offset),
+      default_group_name(std::move(group_name)) {}
+
 void* CUDASymmetricMemoryAllocator::alloc(
     size_t size,
     int device_idx,
-    const std::string& group_name) {
+    const std::optional<std::string>& group_name) {
 #if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
-  auto driver_api = c10::cuda::DriverAPI::get();
+  c10::cuda::CUDAGuard guard(device_idx);
+  device_idx = static_cast<int>(guard.current_device().index());
```

Comment:
What's the point of this?

Reply:
Just wanted to move an existing guard up so that future changes relying on cudaSetDevice become more foolproof.